src/Share/Survey/SurveyHelper.cs

```diff
@@ -78,213 +64,76 @@ public static SurveyHelper GetInstance()
             }
         }
 
-        private ScheduleInfo GetScheduleInfo()
-        { 
-            return new ScheduleInfo() { LastPromptDate = LastPromptDate.ToString("yyyy-MM-dd"), Modules = Modules.Values.ToList() };
-        }
-
-        private bool ReadFromStream()
+        private void InitialSurveyHelper()
         {
-            StreamReader sr = null;
-            try
-            {
-                if (File.Exists(SurveyScheduleInfoFile))
-                {
-                    sr = new StreamReader(new FileStream(SurveyScheduleInfoFile, FileMode.Open, FileAccess.Read, FileShare.None));                    
-                    MergeScheduleInfo(JsonConvert.DeserializeObject<ScheduleInfo>(sr.ReadToEnd()));
-                }
-            }
-            catch (Exception e)
+            if (File.Exists(SurveyScheduleInfoFile))
             {
-                if (e is UnauthorizedAccessException)
-                {
-                    _ignoreSchedule = true;
-                }
-                //deserialize failed, means content of file is incorrect, make file empty
-                if (e is JsonException)
-                {
-                    if (sr != null)
-                    {
-                        sr.Dispose();
-                    }
-                    EmptyFileAsync();
-                }
-                return false;
-            }
-            finally
-            {
-                if (sr != null)
-                {
-                    sr.Dispose();
+                using (StreamReader sr = new StreamReader(new FileStream(SurveyScheduleInfoFile, FileMode.Open, FileAccess.Read, FileShare.None))) {
+                    ScheduleInfo scheduleInfo = JsonConvert.DeserializeObject<ScheduleInfo>(sr.ReadToEnd());
+                    DateTime date = DateTime.MinValue;
+                    DateTime.TryParse(scheduleInfo.LastActiveDay, out date);
+                    LastActiveDay = date;
+                    ActiveDays = scheduleInfo.ActiveDays;
+                    DateTime.TryParse(scheduleInfo.ExpectedDate, out date);
+                    ExpectedDate = date;
+                    DateTime.TryParse(scheduleInfo.LastPromptDate, out date);
+                    LastPromptDate = date;
+                    return;
                 }
             }
-            return true;
+            LastActiveDay = Today;
+            LastPromptDate = Today;
+            ExpectedDate = DateTime.MinValue;
+            ActiveDays = 1;
+            WriteToStream(JsonConvert.SerializeObject(GetScheduleInfo()));
+        }
+
+        public ScheduleInfo GetScheduleInfo()
+        { 
+            return new ScheduleInfo() 
+            { 
+                LastPromptDate = LastPromptDate.ToString("yyyy-MM-dd"), 
+                ActiveDays = ActiveDays, 
+                LastActiveDay = LastActiveDay.ToString("yyyy-MM-dd"),  
+                ExpectedDate = ExpectedDate.ToString("yyyy-MM-dd")
```

Comment:
DateTime.ToString() will use local culture by default, see https://docs.microsoft.com/en-us/dotnet/api/system.datetime.tostring?view=netstandard-2.1#system-datetime-tostring(system-string). 

It may cause exception when DateTime.TryParse() and DateTime.ToString() use different cultures. Please add DateTimeFormatInfo.InvariantInfo when operate DateTime objects.